clang/lib/CIR/CodeGen/CIRGenExprScalar.cpp

```diff
@@ -148,3 +150,27 @@ mlir::Value ScalarExprEmitter::VisitCastExpr(CastExpr *ce) {
   }
   return {};
 }
+
+/// Return the size or alignment of the type of argument of the sizeof
+/// expression as an integer.
+mlir::Value ScalarExprEmitter::VisitUnaryExprOrTypeTraitExpr(
+    const UnaryExprOrTypeTraitExpr *e) {
+  const QualType typeToSize = e->getTypeOfArgument();
+  if (e->getKind() == UETT_SizeOf) {
+    if (const VariableArrayType *variableArrTy =
+            cgf.getContext().getAsVariableArrayType(typeToSize)) {
+      cgf.getCIRGenModule().errorNYI(e->getSourceRange(),
```

Comment:
How is this recovering?  Will there be a known-const-int in this case to give a valid value?  Same on 169.

Reply:
I will check `EvaluateKnownConstInt` and create a dummy value for recovering

Reply:
Nice i used 1 as recovering value for unimplemented yet cases